include/fastdds/statistics/dds/domain/DomainParticipant.hpp

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file DomainParticipant.hpp
+ *
+ */
+
+#ifndef _FASTDDS_STATISTICS_DDS_DOMAIN_DOMAINPARTICIPANT_HPP_
+#define _FASTDDS_STATISTICS_DDS_DOMAIN_DOMAINPARTICIPANT_HPP_
+
+#include <string>
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/publisher/qos/DataWriterQos.hpp>
+#include <fastrtps/fastrtps_dll.h>
+#include <fastrtps/types/TypesBase.h>
+
+using eprosima::fastrtps::types::ReturnCode_t;
+
+namespace eprosima {
+namespace fastdds {
+namespace statistics {
+namespace dds {
+
+/**
+ * Class DomainParticipant: extends standard DDS DomainParticipant class to include specific methods for the Statistics
+ * module
+ * @ingroup STATISTICS_MODULE
+ */
+class DomainParticipant : public eprosima::fastdds::dds::DomainParticipant
```

Comment:
This class is not meant to be directly constructed, so its default constructor should be `=delete`d

Reply:
Done in 80ef7c62e4d554e676f197a8d5a79f55d0577d29